src/agent.rs

```diff
@@ -47,6 +35,8 @@ where
 {
 }
 
+impl <T> Transferable for T where T: Serialize + for<'de> Deserialize<'de> {}
```

Comment:
Given how `Agent`s have to use input and output types that implement `Transferable`, was there any reason why implementation for all types wasn't chosen in the first place? It seems inconvenient for users of Yew to have to specify which types can be transferred to and from `Agent`s. 

Does the compiler emit significantly more code when this is done?

Instead of the changes made in this commit, `Agent`'s trait bounds could instead be relaxed to require something like this 
```rust
pub trait Agent: Sized + 'static {
    /// Reach capaility of the agent.
    type Reach: Discoverer;
    /// Type of an input messagae.
    type Message;
    /// Incoming message type.
    type Input: Serialize + for <'de> Deserialize<'de>;
    /// Outgoing message type.
    type Output: Serialize + for<'de> Deserialize<'de>;
```

instead of this:
```rust
pub trait Agent: Sized + 'static {
    /// Reach capaility of the agent.
    type Reach: Discoverer;
    /// Type of an input messagae.
    type Message;
    /// Incoming message type.
    type Input: Transferable;
    /// Outgoing message type.
    type Output: Transferable;
```